Libplanet/Net/Swarm.BlockCandidate.cs

```diff
@@ -0,0 +1,407 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Libplanet.Blockchain;
+using Libplanet.Blocks;
+
+namespace Libplanet.Net
+{
+    public partial class Swarm<T>
+    {
+        private async Task BlockCandidateConsumer(
+            TimeSpan timeout,
+            CancellationToken cancellationToken)
+        {
+            var checkInterval = TimeSpan.FromMilliseconds(100);
+            while (!cancellationToken.IsCancellationRequested)
+            {
+                if (BlockCandidateTable.Any())
+                {
+                    BlockHeader tipHeader = BlockChain.Tip.Header;
+                    SortedList<long, Block<T>> blocks =
+                        BlockCandidateTable.GetCurrentRoundCandidate(tipHeader);
+                    if (!(blocks is null))
+                    {
+                        var latest = blocks.Last();
+                        _logger.Debug(
+                            "{MethodName} has started. " +
+                            "expert : #{BlockIndex} {BlockHash} " +
+                            "Count of {BlockCandidateTable}: {Count}",
+                            nameof(BlockCandidateConsumer),
+                            latest.Value.Index,
+                            latest.Value.Header,
+                            nameof(BlockCandidateTable),
+                            BlockCandidateTable.Count);
+                        _ = BlockCandidateHandlerAsync(
+                            blocks,
+                            timeout,
+                            cancellationToken);
+                        BlockAppended.Set();
```

Comment:
This event is meant to be triggered when block is appended, but it seems not at this moment.

Reply:
`Swarm<T>.BlockCandidateConsumer()` is append to `BlockChain<T>` its blocks. Do you mean `BlockAppended.Set()` is called when appended every single block?

Reply:
It's triggered after a bulk is appended as I know. But in here, at the moment the event is triggered, it is not sure that the block is really appended or not because `_ = BlockCandidateHandlerAsync(blocks, timeout, cancellationToken);` just spawns the task and does not await it (asynchronous).

Reply:
Aha, I missnamed it. it is not `async`. 

```        
private bool BlockCandidateHandlerAsync(
            SortedList<long, Block<T>> candidate,
            TimeSpan timeout,
            CancellationToken cancellationToken)
```

I'll fix this name and amend it.

Reply:
It's trivial, but `BlockCandidateHandler` listens like it blocks something, as method names usually start with a verb (not a noun).

Reply:
Ooops I see, I should've review more carefully ;)